packages/mongodb-rag-ingest/src/sources/mongodb-university/MongoDbUniversityDataApiClient.test.ts

```diff
@@ -10,17 +10,36 @@ describe("makeMongoDbUniversityDataApiClient()", () => {
   });
   it("should return a MongoDbUniversityDataApiClient", () => {
     expect(mongodbUniversityDataApiClient).toBeDefined();
-    expect(mongodbUniversityDataApiClient).toHaveProperty("getAllCatalogItems");
+    expect(mongodbUniversityDataApiClient).toHaveProperty("getCatalogItems");
     expect(mongodbUniversityDataApiClient).toHaveProperty("getAllVideos");
   });
-  it("should load all the catalog items from the MongoDB University Data API", async () => {
-    const allTiCatalogItems =
-      await mongodbUniversityDataApiClient.getAllCatalogItems();
-    expect(allTiCatalogItems.data).toBeDefined();
-    expect(allTiCatalogItems.metadata).toBeDefined();
-    expect(allTiCatalogItems.metadata.has_more).toBe(false);
-    expect(allTiCatalogItems.data.length).toBeGreaterThan(0);
-  });
+  describe("getCatalogItems()", () => {
+    it("should load all the catalog items from the MongoDB University Data API", async () => {
+      const allTiCatalogItems =
+        await mongodbUniversityDataApiClient.getCatalogItems();
+      expect(allTiCatalogItems.data).toBeDefined();
+      expect(allTiCatalogItems.metadata).toBeDefined();
+      expect(allTiCatalogItems.metadata.has_more).toBe(false);
+      expect(allTiCatalogItems.data.length).toBeGreaterThan(0);
+    });
+    it("getCatalogItems should accept query params", async () => {
+      const tiCatalogItems =
+        await mongodbUniversityDataApiClient.getCatalogItems({
+          public_only: true,
+          learning_formats: ["Learning Path", "Course"],
+          nest_associated_content: true,
```

Comment:
camelCase!

```suggestion
          publicOnly: true,
          learningFormats: ["Learning Path", "Course"],
          nestAssociatedContent: true,
```

address throughout PR